intrinsic_alignments/choose_ia/choose_ia_params.py

```diff
@@ -0,0 +1,15 @@
+from cosmosis.datablock import option_section, names
+
+def setup(options):
+    suffix = options.get_string(option_section, "suffix", "")
+    return suffix
+
+def execute(block, config):
+    suffix = config
+
+    for sec,name in block.keys():
```

Comment:
You can use `block.keys('intrinsic_alignment_parameters')` to avoid having to check the section. Will make it many times faster.  Also you should usually use `and` instead of `&` for logical comparisons, though it doesn't actually matter here.

Reply:
Well spotted - done!